inginious/frontend/plugins/task_hints/pages/user_hint_manager.py

```diff
@@ -163,6 +188,119 @@ def update_total_penalty(self, task_id, username):
+                    
+                    if group_hints_document: 
+                    
+                        group_hints_document_members = group_hints_document.get("username")
+
+                        for member in group_hints_document_members:
+                            if member not in students:
+
+                                self._database.user_hints.find_one_and_update(
+                                    {"taskid": task_id, "username": students},
+                                    {"$pull": {
+                                        "username": member
+                                        }
+                                    }
+                                )
+                                self.insert_default_user_hints(task_id, member)
+                                
+                    else: 
+                        group_user_hints = []
+                        for student in students:
+                            user_hints = self.get_user_hints(task_id, student)
+                            if user_hints: 
+                                user_hints_id = user_hints.get("_id")
+                                
+                                group_user_hints.append(user_hints)
+
+                                self._database.user_hints.find_one_and_update(
+                                    {"_id": user_hints_id},
+                                    {"$pull": {
+                                            "username": student
+                                        }
+                                    }
+                                )
+
+
+                        common_user_hints = set()
+                        if len(group_user_hints) > 1:
+                        
+                            common_user_hints = set([hint.get("id") for hint in group_user_hints[0].get("unlocked_hints")])
+                            for document in group_user_hints[1:]:
+
+                                temp = set([hint.get("id") for hint in document.get("unlocked_hints")])
+                                common_user_hints = common_user_hints.intersection(temp)
+
+                        
+                        common_user_hints_data = []
+                        for hint in task_hints.values():
+                            if hint['id'] in common_user_hints:
+                                common_user_hints_data.append({"id": hint['id'],"penalty": hint['penalty']})
+
+                        if common_user_hints_data:
+                            self._database.user_hints.insert(
+                                {"taskid": task_id, "username": students, "unlocked_hints": common_user_hints_data, "total_penalty": 0})
+                            
+                            self.update_total_penalty(task_id, students)
+                                  
+
+        else:
+            
+            classrooms = self._database.aggregations.find({"courseid": courseid})
```

Comment:
This call is done two here and in after the `if` you can do it before the `if` for both cases. Also, the same, this query returns always either nothing or one document, so there is no more than 1 document here.